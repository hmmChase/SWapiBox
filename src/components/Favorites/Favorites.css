button {
  font-family: 'Exo 2', sans-serif;
  font-weight: 600;
  background: none;
  border: 2px solid yellow;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;
  border-radius: 10px;
  color: #fff;
}

button:hover {
  background: rgba(164, 164, 0, 0.5);
}