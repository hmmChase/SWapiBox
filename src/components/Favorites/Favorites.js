import React from 'react';
import PropTypes from 'prop-types';
import './Favorites.css';

const Favorites = props => {
  return (
    <button onClick={() => props.setFavorites(props.category)}>
      Favorites: <span>{props.favorites.length}</span>
    </button>
  );
};

Favorites.propTypes = {
  category: PropTypes.string,
  setFavorites: PropTypes.func,
  favorites: PropTypes.array
};

export default Favorites;
