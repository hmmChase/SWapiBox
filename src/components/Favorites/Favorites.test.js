import React from 'react';
// import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Favorites from './Favorites';

describe('Favorites', () => {
  let favorites;

  beforeEach(() => {
    favorites = shallow(<Favorites />);
  });

  it('matches the snapshot', () => {
    expect(favorites).toMatchSnapshot();
  });

});