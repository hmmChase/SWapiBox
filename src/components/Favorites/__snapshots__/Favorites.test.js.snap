// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    category="mock category"
    favorites={Array []}
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Favorites: ",
        <span />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Favorites: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Favorites: ",
          <span />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Favorites: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Favorites matches the snapshot with favorites added 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    category="mock category"
    favorites={Array []}
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Favorites: ",
        <span />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Favorites: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Favorites: ",
          <span />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Favorites: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
